### PR TITLE
Fix gmaps deprecation modal

### LINKED ISSUE #4926: Google deprecation dialog is not working
Seems like it appears but the the "ok, proceed" button is not working and the dialog doesn't disappear.

image001-1 jpg 01d0d069 d3334340

### PR DESCRIPTION
**Pull request by viddo**
Fixes #4926 

cc @xavijam

### CODE REVIEW

File: lib/assets/javascripts/cartodb/table/mapview.js [hunk 1]
```diff
@@ -378,6 +378,7 @@ cdb.admin.MapTab = cdb.core.View.extend({
     dialog.ok = function() {
       self.map.set('provider', 'leaflet', { silent: true });
       self.setupMap();
+      this.close && this.close();
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/table/mapview.js]: I feel you need to use ```self``` instead of this.

Comment (viddo) [inline lib/assets/javascripts/cartodb/table/mapview.js]: No, because `this` refers to the modal view instance, self refers to the `mapTab` view.

Comment (xavijam) [inline lib/assets/javascripts/cartodb/table/mapview.js]: Ahhh, you are right my friend, then let's go.

**Comment by xavijam** [comment]
:es:

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5651/))

**Pull request merged**